samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1430 @@
+		{
+			req_mem_page_desc.x = std::min((mem_page_desc.x * 2U) + x, virtual_texture.mip_properties[req_mem_page_desc.mip_level].num_columns - 1U);
+			req_mem_page_desc.y = std::min((mem_page_desc.y * 2U) + y, virtual_texture.mip_properties[req_mem_page_desc.mip_level].num_rows - 1U);
+
+			size_t page_index = get_page_index(req_mem_page_desc);
+
+			virtual_texture.page_table[page_index].gen_mip_required = true;
+
+			if (!virtual_texture.page_table[page_index].valid)
+			{
+				mipgen_required_vec.push_back(req_mem_page_desc);
+				virtual_texture.update_set.insert(page_index);
+			}
+		}
+	}
+}
+
+/**
+ * 	@brief Convert information from BLOCK-based into PAGE-based data. BLOCKS are just the abstraction units described by ON_SCREEN_HORIZONTAL_BLOCKS and ON_SCREEN_VERTICAL_BLOCKS. PAGES are the actually allocated chunks of memory, their size is device-dependent.
+ */
+std::vector<size_t> SparseImage::get_memory_dependency_for_the_block(size_t column, size_t row, uint8_t mip_level)
+{
+	std::vector<size_t> dependencies;
+	double              height_on_screen_divider = 1.0 / on_screen_num_vertical_blocks;
+	double              width_on_screen_divider  = 1.0 / on_screen_num_horizontal_blocks;
+
+	double xLow  = width_on_screen_divider * column;
+	double xHigh = width_on_screen_divider * (column + 1U);
+
+	double yLow  = height_on_screen_divider * row;
+	double yHigh = height_on_screen_divider * (row + 1U);
+
+	double texel_width  = virtual_texture.mip_properties[mip_level].width;
+	double texel_height = virtual_texture.mip_properties[mip_level].height;
+
+	double in_memory_row_pages    = texel_height / virtual_texture.format_properties.imageGranularity.height;
+	double in_memory_column_pages = texel_width / virtual_texture.format_properties.imageGranularity.width;
+
+	double height_in_memory_divider = 1.0 / in_memory_row_pages;
+	double width_in_memory_divider  = 1.0 / in_memory_column_pages;
+
+	size_t memXLow  = static_cast<size_t>(floor(xLow / width_in_memory_divider));
+	size_t memXHigh = static_cast<size_t>(ceil(xHigh / width_in_memory_divider));
+
+	size_t memYLow  = static_cast<size_t>(floor(yLow / height_in_memory_divider));
+	size_t memYHigh = static_cast<size_t>(ceil(yHigh / height_in_memory_divider));
+
+	for (size_t y = memYLow; y < memYHigh; y++)
+	{
+		for (size_t x = memXLow; x < memXHigh; x++)
+		{
+			size_t page_index = virtual_texture.mip_properties[mip_level].mip_base_page_index + virtual_texture.mip_properties[mip_level].num_columns * y + x;
+			dependencies.push_back(page_index);
+		}
+	}
+	return dependencies;
+}
+
+/**
+ * 	@brief Compare required and currently present mip_level for each BLOCK.
+ */
+void SparseImage::compare_mips_table()
+{
+	if (!virtual_texture.texture_block_update_set.empty())
+	{
+		virtual_texture.texture_block_update_set.clear();
+	}
+	for (size_t y = 0U; y < virtual_texture.current_mip_table.size(); y++)
+	{
+		for (size_t x = 0U; x < virtual_texture.current_mip_table[0].size(); x++)
```

Comment:
Maybe add a short comment here, that all current_mip_table[i] have the same size. Or just use `current_mip_table[i].size()` here.